### PR TITLE
metadata scan

### PR DESCRIPTION
**Pull request by dongdongyang33**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #8342

## What this PR does / why we need it:
add metadata scan (table function version)

BVT cases only cover invariant result columns like col_name/rows_count/null_count/compress_size/origin_size. The variable result columns like block_id are not covered

### CODE REVIEW

**Review by fengttt — COMMENTED** [review]
this feature definitely need bvt test cases.

**Comment by dongdongyang33** [comment]
> this feature definitely need bvt test cases.

Yes, I will add it.

**Review by nnsgmsone — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by m-schen — APPROVED** [review]
colexec and function lgtm.

File: pkg/sql/plan/metadata_scan.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	"github.com/matrixorigin/matrixone/pkg/sql/parsers/tree"
+)
+
+var (
+	MetadataScanColDefs = []*plan.ColDef{
+		{
+			Name: catalog.MetadataScanInfoNames[catalog.COL_NAME],
+			Typ: &plan.Type{
+				Id:          int32(catalog.MetadataScanInfoTypes[catalog.COL_NAME].Oid),
+				NotNullable: false,
```
Comment (aunjgr) [inline pkg/sql/plan/metadata_scan.go:29]: Can these `NotNullable`s be set to true?

**Review by reusee — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by qingxinhome — APPROVED** [review]

**Review by lacrimosaprinz — APPROVED** [review]

**Pull request merged**